source/guides/network/wifi.rst

```diff
@@ -102,8 +107,149 @@ TUI (Text-based User Interface)
 
    .. figure:: /_figures/wifi/nmtui_5.png
 
+Using wpa_supplicant
+********************
+
+Make sure NetworkManager is either stopped, disabled by maksing the service,
+or not installed before using wpa_supplicant.
+
+.. code-block:: bash
+
+   sudo systemctl stop NetworkManager.service
+   sudo systemctl mask NetworkManager.service
+
+#. Create a ``wpa_supplicant`` configuration directory.
+
+   .. code-block:: bash
+
+      sudo mkdir /etc/wpa_supplicant
+
+#. Determine your wireless interface name.
+
+   .. code-block:: bash
+
+      iw dev
+
+   Use the name following "Interface" on the first line (eg. wlp1s0)
+
+   .. code-block:: console
+
+       Interface wlp1s0
+          ifindex 3
+          wdev 0x1
+          addr 00:xx:xx:38:34:7a
+          type managed
+          txpower 0.00 dBm
+
+   Set the $INTERFACE_NAME environment variable to take advantage of copying
+   and pasting commands.
+
+   .. code-block:: bash
+
+      export INTERFACE_NAME=wlp1s0
+
+#. Create a minimal configuration file called
+   :file:`/etc/wpa_supplicant/wpa_supplicant-$INTERFACE_NAME.conf`
+   and add the following:
+
+   .. code-block:: bash
+
+      ctrl_interface_group=wheel
+      ctrl_interface=/run/wpa_supplicant
+      update_config=1
+
+#. Complete the configuration process.
+
+   .. code-block:: bash
+
+      sudo wpa_supplicant -B -i $INTERFACE_NAME -c /etc/wpa_supplicant/wpa_supplicant-$INTERFACE_NAME.conf
+
+#. Use :command:`wpa_cli` (interactive mode) to scan for available networks.
+
+   .. code-block:: bash
+
+      sudo wpa_cli
+      > scan
+      OK
+      <3>CTRL-EVENT-SCAN-STARTED
+      <3>CTRL-EVENT-SCAN-RESULTS
+      > scan_results
+      bssid / frequency / signal level / flags / ssid
+      00:xx:xx:73:7b:46 5180 -55 [WPA2-PSK-CCMP][ESS] Network1
+      00:xx:xx:5d:d9:23 2412 -47 [RSN--CCMP][MESH] 1137e9
+      00:xx:xx:73:7b:43 2412 -49 [RSN--CCMP][MESH] 1137e9
+      00:xx:xx:5d:d9:25 2412 -47 [WPA2-PSK-CCMP][ESS] Network1
+      00:xx:xx:37:25:05 2412 -49 [WPA2-PSK-CCMP][ESS] Network1
+      00:xx:xx:73:7b:45 2412 -59 [WPA2-PSK-CCMP][ESS] Network1
+      00:xx:xx:83:fa:6a 2437 -57 [WPA-EAP-CCMP+TKIP][WPA2-EAP-CCMP+TKIP][ESS]
+      00:xx:xx:83:fa:70 5240 -76 [WPA2-EAP-CCMP][ESS] Network2
+      00:xx:xx:4f:e9:2c 2412 -67 [WPA2-PSK-CCMP][ESS][P2P] Printer
+      00:xx:xx:af:fe:3e 5765 -79 [WPA2-PSK-CCMP][ESS] Network3
+      00:xx:xx:e9:eb:29 2412 -76 [WPA2-PSK-CCMP][ESS] Network4
+      00:xx:xx:26:4a:b9 2412 -79 [WPA2-PSK-CCMP][ESS][P2P] Printer2
+      00:xx:xx:b9:0d:d4 2462 -79 [WPA2-PSK-CCMP][ESS] Network5
```

Comment:
@puneetse. Can you verify that you tested this section until the end for validity? 
I reached this section without issues, yet after entering `scan` in the `wpa_cli`, the stdout hangs at: 
`<3>CLTRL-EVENT-NETWORK-NOT-FOUND` ... 

Eventually, the initial scan returns `OK`. 
But it still hangs. I don't believe this is a proxy issue. 

I'm alright

Reply:
@mvincerx I'm pretty sure it doesn't hang, the prompt probably just didn't return right away. Hit return and continue.